### ISSUE TITLE
Add @ExpectMaxUpdate

### ISSUE THREAD

**Issue by jeanbisutti** [quick-perf/quickperf#68]
**Description**

The aim of this annotation is to check the maximum number of UPDATE statements.

For example, a test method having `@ExpectMaxUpdate(2)` annotation will fail if more than two UPDATE statements are executed.

**Implementation steps**

1) Create the `@ExpectMaxUpdate` annotation [here](https://github.com/quick-perf/quickperf/tree/master/sql/sql-annotations/src/main/java/org/quickperf/sql/annotation)

    Example: [@ExpectUpdate](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/annotation/ExpectUpdate.java)

2) Create a test class in [this package](https://github.com/quick-perf/quickperf/tree/master/sql/sql-integration-test/src/test/java) and add automatic tests

    Example: [ExpectUpdateTest.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-integration-test/src/test/java/ExpectUpdateTest.java)

3) In [this package](https://github.com/quick-perf/quickperf/tree/master/sql/sql-annotations/src/main/java/org/quickperf/sql/update) add a performance measure extractor and a performance issue verifier. 

    Examples: 
    * [UpdateCountMeasureExtractor.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/update/UpdateCountMeasureExtractor.java)
    * [UpdateNumberPerfIssueVerifier.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/update/UpdateNumberPerfIssueVerifier.java)  
     * [MaxOfSelectsPerfIssueVerifier.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/select/MaxOfSelectsPerfIssueVerifier.java)

4) Create an AnnotationConfig in [SqlAnnotationsConfig](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/config/library/SqlAnnotationsConfigs.java). This new AnnotationConfig will use the previously created performance measure extractor, performance issue verifier together with a PersistenceSqlRecorder.

    Example:
    ```java
        static final AnnotationConfig MAX_SQL_SELECT = new AnnotationConfig.Builder()
            .perfRecorderClass(PersistenceSqlRecorder.class)
            .perfMeasureExtractor(SelectCountMeasureExtractor.INSTANCE)
            .perfIssueVerifier(MaxOfSelectsPerfIssueVerifier.INSTANCE)
            .build(ExpectMaxSelect.class);
     ```

5) Use the new AnnotationConfig in `loadAnnotationConfigs()` method of [SqlConfigLoader](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/config/library/SqlConfigLoader.java) 

6) Complete the measure extractor and the performance issue verifier to make the tests green. Refactor to clean the code.

7) In [SqlAnnotationBuilder](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/annotation/SqlAnnotationBuilder.java), add a static method to can build the new annotation. This method can be used to define the annotation with [a global scope](https://github.com/quick-perf/doc/wiki/QuickPerf#annotation-scopes).

If you start to work on this issue, please leave a comment _I start to work on this issue_, so we can assign it to you. Do not hesitate to ask for information about the issue.

**Comment by FTarfasse** [comment]
Hi Jean,

I'd really like to give this a try if it's ok for you. To be perfectly honest it will take me some time as I am on the beginner side but it's a perfect occasion to better understand your nice tool !

If I have any questions should I reply on this same thread ?
Thanks,

**Comment by jeanbisutti** [comment]
Hi Fabrice,

Yes, you can work on this issue! I assigned it to you.

Don't hesitate to leave of comment on this issue if you need help.

**Comment by jeanbisutti** [comment]
Fixed with  f79ca2e46b1d0a29d46f4d44a3a037325a12ca6e.
